Add BBEs for the persistence feature (#4429)
ballerina-platform/ballerina-distribution: examples/persist-filtering/persist_filtering.md

> # Persist read - Filtering and sorting

Reviewer: if we rename the BBE titles as per the previous comments in index json, fix these titles and URL paths too to match with that.

> The Ballerina persistence feature provide support to manage data persistence in a Ballerina package. It starts with defining the application's data model. Once model is defined, the client API is generated with resources based on the model. The generated

Reviewer 2: ```suggestion
The bal persist feature provide support to manage data persistence in a Ballerina package. It starts with defining the application's data model. Once model is defined, the client API is generated with resources based on the model. The generated
```

Reviewer 2: And check other places

> The Ballerina persistence feature provide support to manage data persistence in a Ballerina package. It starts with defining the application's data model. Once model is defined, the client API is generated with resources based on the model. The generated
> API can be used to query and manipulate the persistent data in the application.
> The generated client API provides `get` resource function to retrieve all records from the data store. As the `get` resource function returns a stream of records, we can use query expressions to do additional filter. For more information, see [Query Expressions](/learn/by-example/query-expressions/).

Reviewer 3: ```suggestion
The Ballerina persistence feature provides support to manage data persistence in a Ballerina package. It starts with defining the application's data model. Once the model is defined, the client API is generated with resources based on the model. The generated
API can be used to query and manipulate the persistent data in the application.
The generated client API provides  a `get` resource function to retrieve all records from the data store. As the `get` resource function returns a stream of records, we can use query expressions to perform additional filters. For more information, see [Query Expressions](/learn/by-example/query-expressions/).
```

> > **Note:** This example uses the Ballerina tables as the data store. You can MySQL and Google Sheets as the data store as well. For more information, see [Supported Data Stores](/learn/supported-data-stores/).

Reviewer 3: ```suggestion
> **Note:** This example uses Ballerina tables as the data store. You can use MySQL and Google Sheets as the data store as well. For more information, see [Supported Data Stores](/learn/supported-data-stores/).
```
